mempool/policy.go

```diff
@@ -226,8 +214,15 @@ func isDust(txOut *wire.TxOut, minRelayTxFee btcutil.Amount) bool {
 	//
 	// The most common scripts are pay-to-pubkey-hash, and as per the above
 	// breakdown, the minimum size of a p2pkh input script is 148 bytes.  So
-	// that figure is used.
-	totalSize := txOut.SerializeSize() + 148
+	// that figure is used. If the output being spent is a witness program,
+	// then we apply the witness discount to the size of the signature.
+	var totalSize int
```

Comment:
Both paths use the `SerializeSize` as the base as well as the non-discounted portions of the input.  Thus, I'd suggest the following which I think is easier to follow:

```Go
totalSize := txOut.SerializeSize() + 41
if txscript.IsWitnessProgram(txOut.PkScript) {
	totalSize += (107 / blockchain.WitnessScaleFactor)
} else {
	totalSize += 107
}
```

Also, please update the comments to include the breakdown for the segwit script in question as well as the fact the discount is applied to the input script (witness data) for witness programs so it's clear how the numbers are derived.